### ISSUE TITLE
debugger uses lowercase drive letter for script path

### ISSUE THREAD

**Issue by fireattack** [microsoft/debugpy#929]

**Comment by fabioz** [comment]
Also, note that having something uppercase isn't really a standard in windows (it's just how you did `cd` into the directory in your shell... you can also `cd` into a directory lowercase and it'll be kept as lowercase -- keep in mind that if you have a `.pyc` this information may be saved into the `.pyc` and you may need to erase it to update).

**Comment by fireattack** [comment]
Thanks for your insight! 

Do you think the bug/issue around substitution of ${file} into lower case drive letter mentioned above part of `vscode-python`, or even `vscode` proper?

As you can see, I did open this ticket initially in `microsoft/vscode-python`.

**Comment by fabioz** [comment]
@fireattack the `${file}` substitution is tackled at `vscode` itself, whereas the shortcut to launch it is handled at `vscode-python`. As a note, I'm not convinced myself that using an upper-case drive letter is the best approach (as I mentioned, this is not a consensus).

I'm curious about your use case though, why does it matter if it's upper case or lower case? I think you program ought to handle both anyways as failing to do that in your program would probably be a bug...

**Comment by fireattack** [comment]
You're right, I probably should handle it better in my program. But here is the gist of it:

I have a huge amount of file `Path`s and want to quickly check if they are inside within certain dir (not necessarily directly) .

So I check this way:
```python
return str(filepath).startswith(str(rootdir))`
```

In my testing, this is much faster than:
```python
return any(parent == rootdir for parent in a.parents)
```
Which obviously is more robust and won't be affected by lowercase drive letter.

The reason why `${file}` substitution affects my case is because I get these `Path`s or `rootdir` via `Path(__file__).parent.parent / 'datafolder'` since my whole working dir isn't in a fixed folder.

**Comment by fabioz** [comment]
Well, presumably the `rootdir` is fixed, so, you could do:

`rootdir = str(rootdir).lower()`

Earlier, and then use something as:

`return str(filepath).lower().startswith(rootdir)`

afterwards (which shouldn't be such a big performance hit).

I can think of other ways where this may still fail, but at least the issue with the casing will be fixed ;)

**Comment by fireattack** [comment]
Yeah, that works. I found another solution is to use `Path.resolve()`: which will always gives you upper case letter for drive letter. I can use it onPath(__file__) to "fix" it, before I'm trying to grab any paths from it.

**Comment by fabioz** [comment]
As a note, `resolve()` may be a footgun because it'll also resolve substs and links -- so, if you have a `substs` `w:` which points to `c:\temp`, it'll be resolved to `c:\temp` (so, you can use it but beware that if you create links/substs it may not do exactly what you want).
